### PR TITLE
Add perltidy formatter and perl samples

### PR DESCRIPTION
**Pull request by aborazmeh**

### CODE REVIEW

File: plugin/defaults.vim [hunk 1]
```diff
@@ -211,3 +211,18 @@ endif
 if !exists('g:formatters_dart')
     let g:formatters_dart = ['dartfmt']
 endif
+
+" Perl
+if !exists('g:formatdef_perltidy')
+  " use perltidyrc file if readable
+  if (has("win32") && (filereadable("perltid.ini") || filereadable($HOMEPATH."/perltidy.ini"))) ||
```
Comment (Chiel92) [inline plugin/defaults.vim]: Is `"perltid.ini"` (without the `y`) a typo or is this intended?

**Comment by aborazmeh** [comment]
A typo!

**Comment by Chiel92** [comment]
This is the first formatter that checks for existing config file, which is a good thing in my opinion!
I think it would be even better if it were done on-the-fly.
This is because during a vim session the existence of config files or the current directory may change.

To this end you would move the logic into a function and call this function from within the formatter definition expression. Would you mind giving that a try?

**Comment by Chiel92** [comment]
Apart from what I just said, it seems like a sound definition, so I'm merging it in anyway. Thanks for that!

**Pull request merged**